lib/services/plugins-service.ts

```diff
@@ -0,0 +1,48 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import path = require("path");
+import shelljs = require("shelljs");
+import constants = require("./../constants");
+
+export class PluginsService implements IPluginsService {
+	constructor(private $npm: INodePackageManager,
+		private $fs: IFileSystem,
+		private $projectData: IProjectData,
+		private $platformsData: IPlatformsData,
+		private $childProcess: IChildProcess) { }
+		
+	public get modulesDestinationPath() {
+		let platformData = this.$platformsData.getPlatformData("ios");
+		return path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME, "tns_modules");
+	} 
+		
+	public add(pluginName: string): IFuture<void> {
+		return (() => {
+			let command = `npm install ${pluginName} --save`;
```

Comment:
Do you think it is a good idea to support --production flag for tns plugin add as well? It will simply be passed to npm.
At any rate this functionality is for another PR.